### PR TITLE
Windows & QOL

### PR DESCRIPTION
**Pull request by nuIIpointerexception**
Fix windows binary download and unzipping.
Improve README.md and fix typo using harper, lol.
Add more officially supported languages by zed to the extension.

### CODE REVIEW

**Comment by nuIIpointerexception** [comment]
I mean considering they don't have support for many languages yet. Maybe it'll still be useful to add, no need to update because harper might add them in the future?

File: src/lib.rs [hunk 1]
```diff
@@ -1,129 +1,155 @@
...
-                let entry = entry.map_err(|e| format!("failed to load directory entry {e}"))?;
-                if entry.file_name().to_str() != Some(&version_dir) {
-                    fs::remove_dir_all(entry.path()).ok();
+            if let Ok(entries) = fs::read_dir(".") {
+                for entry in entries.flatten() {
+                    if entry.path() != version_dir {
+                        fs::remove_dir_all(entry.path()).ok();
+                    }
                 }
             }
```
Comment (Stef16Robbe) [inline src/lib.rs:123]: This change contains a regression for MacOS. On a fresh install, this new approach will remove the binary location. I haven't looked into why. The version dir will still exist, just without the LSP binary in it.

If this is a fix you made for Windows we'll have to investigate how to fix it for other platforms, if it's purely stylistically I prefer to revert it.

**Review by Stef16Robbe — CHANGES_REQUESTED** [review]
Thanks! Please refer to the regression.

File: src/lib.rs (hunk 1 above)
Comment (nuIIpointerexception) [reply src/lib.rs:123]: Im pretty sure we can revert it. this is just stylistically and will not do anything for the windows fix

Comment (nuIIpointerexception) [reply src/lib.rs]: Alright so i pushed a fix, could you test the extension now?
I do not have a access to a mac, but i don't think this shouldn't fail now.

Comment (Stef16Robbe) [reply src/lib.rs]: Tested, works thanks!

Comment (Stef16Robbe) [reply src/lib.rs]: Waiting for merge in extensions repo: https://github.com/zed-industries/extensions/pull/1699

**Comment by Stef16Robbe** [comment]
@nuIIpointerexception PR has been made: https://github.com/zed-industries/extensions/pull/1699

**Pull request merged**